integrations/jira/jira/client.py

```diff
@@ -145,3 +153,41 @@ async def get_paginated_issues(self) -> AsyncGenerator[list[dict[str, Any]], Non
             issue_response_list = (await self._get_paginated_issues(params))["issues"]
             yield issue_response_list
             params["startAt"] += PAGE_SIZE
+
+    async def get_paginated_users(
+        self,
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        logger.info("Getting users from Jira")
+
+        params = self._generate_base_req_params()
+
+        total_users = len(await self._get_users_data(params))
+
+        if total_users == 0:
+            logger.warning(
+                "User query returned 0 users, did you provide the correct Jira API credentials?"
+            )
+
+        params["maxResults"] = PAGE_SIZE
+        while params["startAt"] < total_users:
+            logger.info(f"Current query position: {params['startAt']}/{total_users}")
+
+            user_response_list = await self._get_users_data(params)
+
+            if not user_response_list:
+                logger.warning(f"No users found at {params['startAt']}")
+                break
+
+            logger.info(
+                f"Retrieved users: {len(user_response_list)} "
+                f"(Position: {params['startAt']}/{total_users})"
+            )
+
+            yield user_response_list
+            params["startAt"] += PAGE_SIZE
+            
+    async def get_single_user(self, account_id: str) -> dict[str, Any]:
+        user_response = await self.client.get(f"{self.api_url}/user", params={"accountId": account_id})
+        user_response.raise_for_status()
```

Comment:
I think it will be great if we reuse the _get_user_date() you created above and pass the params={"accountId": account_id}